Gem/Code/Source/AutoGen/NetworkPlayerMovementComponent.AutoComponent.xml

```diff
@@ -33,6 +33,30 @@
     <NetworkInput Type="bool"      Name="Crouch"      Init="false" />
     <NetworkInput Type="uint8_t"   Name="ResetCount"  Init="0" />
 
+    <RemoteProcedure
+        Name="ApplyImpulse"
+        InvokeFrom="Server"
+        HandleOn="Authority"
+        IsPublic="false"
+        IsReliable="true"
+        GenerateEventBindings="true"
+        Description="Applies an impulse directly to the players self generated velocity">
+        <Param Type="AZ::Vector3" Name="Impulse" />
+        <Param Type="bool" Name="External" Description="If true, the impulse will be applied to the external velocity, and if false it will be applied to internal velocity" />
+    </RemoteProcedure>
+
+    <RemoteProcedure
+        Name="SetVelocity"
+        InvokeFrom="Server"
+        HandleOn="Authority"
+        IsPublic="false"
+        IsReliable="true"
+        GenerateEventBindings="true"
+        Description="Directly sets the self generated velocity to the provided value, overriding any current velocity values">
+        <Param Type="AZ::Vector3" Name="Velocity"/>
+        <Param Type="bool" Name="External" Description="If true, the impulse will be applied to the external velocity, and if false it will be applied to internal velocity" />
+    </RemoteProcedure>
```

Comment:
Nit: The bool is probably unnecessary, since anything wanting to set the player velocity through an RPC is "external" by definition i would think, so it always ought to be true. 

Not harmful to have it here though, other than more network data and needing to look up what "true" means when looking at the calling code.